polaris/hub/client.py

```diff
@@ -174,6 +178,20 @@ def _normalize_owner(
 
         return artifact_owner if isinstance(artifact_owner, HubOwner) else HubOwner(slug=artifact_owner)
 
+    @staticmethod
+    def _normalize_verify_checksum(
+        verify_checksum: Optional[bool],
+        dataset: Dataset,
+    ):
+        if verify_checksum is not None:
+            return verify_checksum
+        return dataset._md5sum is not None and not dataset.uses_zarr
```

Comment:
**Comment**
I'm not liking direct access of private properties on the dataset object. If the intent is to check that the md5sum exists, then I'd suggest adding another property to the dataset class to expose this.